lib/src/screens/network_selection/widgets/network_list.dart

```diff
@@ -25,7 +27,13 @@ class NetworkList extends StatelessWidget {
         itemBuilder: (context, index) {
           final network = networks[index];
           return GestureDetector(
-            onTap: () => onSelectNetwork(network),
+            onTap: () async {
+              if (network.security == '-') {
```

Comment:
nit - you could use your helper:
```
bool isPublicNetwork(NetworkInfo network) {
    return network.security == '-';
  }
```
in a few more places (like here) if it were an extension on `NetworkInfo`
```
extension on NetworkInfo {
  bool get isPublicNetwork {
    return security == '-';
  }
}
```